var/spack/repos/builtin/packages/py-ray/package.py

```diff
@@ -36,6 +53,12 @@ class PyRay(PythonPackage):
     depends_on('py-py-spy@0.2.0:', type=('build', 'run'))
     depends_on('py-pyyaml', type=('build', 'run'))
     depends_on('py-requests', type=('build', 'run'))
-    depends_on('py-redis@3.3.2:3.4', type=('build', 'run'))
+    depends_on('py-redis@3.3.2:3.4', type=('build', 'run'), when='@:1.0.1')
+    depends_on('py-redis@3.5.0:4', type=('build', 'run'), when='@1.1.0:')
```

Comment:
```suggestion
    depends_on('py-redis@3.5.0:3', type=('build', 'run'), when='@1.1.0:')
```
Version ranges are inclusive